avatars/processors/inter_record_range_difference.py

```diff
@@ -0,0 +1,265 @@
+    - a variable containing the range between the start and end variables
+    (i.e. `var_b_t` - `var_a_t`)
+
+    This processor is not suitable for data where any of the variables passed as args contain
+    missing values.
+
+    Keyword Arguments
+    -----------------
+        id_variable:
+            variable indicating which individual each row belongs to
+        target_start_variable:
+            variable representing the start of the range to transform
+        target_end_variable:
+            variable representing the end of the range to transform
+        sort_by_variable_name:
+            variable used to sort records for each id
+        new_first_variable_name:
+            name of the variable to be created to contain the first value of the target variable
+        new_range_variable_name:
+            name of the variable to be created to contain the range value
+        new_difference_variable_name:
+            name of the variable to be created to contain the difference value
+
+    Examples
+    --------
+    >>> df = pd.DataFrame(
+    ...     {
+    ...         "id": [1, 2, 1, 1, 2, 2],
+    ...         "start": [7, 14, 6, 12, 10, 23],
+    ...         "end": [8, 18, 7, 15, 12, 24],
+    ...     }
+    ... )
+    >>> processor = InterRecordRangeDifferenceProcessor(
+    ...     id_variable="id",
+    ...     target_start_variable='start',
+    ...     target_end_variable='end',
+    ...     sort_by_variable_name="start",
+    ...     new_first_variable_name="first_value",
+    ...     new_range_variable_name='range_value',
+    ...     new_difference_variable_name="value_difference",
+    ... )
+    >>> processor.preprocess(df)
+       id  range_value  first_value  value_difference
+    0   1            1            6               0.0
+    1   2            4           10               2.0
+    2   1            1            6               0.0
+    3   1            3            6               4.0
+    4   2            2           10               0.0
+    5   2            1           10               5.0
+
+    The postprocess allows you to transform some preprocessed data back into its original format
+
+    >>> preprocessed_df = pd.DataFrame(
+    ...     {
+    ...         "id": [1, 2, 1, 1, 2, 2],
+    ...         "range_value": [1, 4, 1, 3, 2, 1],
+    ...         "first_value": [6, 10, 6, 6, 10, 10],
+    ...         "value_difference": [0, 2, 0, 4, 0, 5],
+    ...     }
+    ... )
+    >>> processor.postprocess(df, preprocessed_df)
+       id  start   end
+    0   1    7.0   8.0
+    1   2   14.0  18.0
+    2   1    6.0   7.0
+    3   1   12.0  15.0
+    4   2   10.0  12.0
+    5   2   23.0  24.0
+    """
+
+    def __init__(
+        self,
+        *,
+        id_variable: str,
+        target_start_variable: str,
+        target_end_variable: str,
+        sort_by_variable_name: str,
+        new_first_variable_name: str,
+        new_range_variable_name: str,
+        new_difference_variable_name: str,
```

Comment:
All of these are expressing a variable name, they should all have the same syntax. Either they all end with `_name` or None of them do.